specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/DaprComponents.json

```diff
@@ -0,0 +1,356 @@
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": false,
+        "x-ms-examples": {
+          "Create or update dapr component": {
+            "$ref": "./examples/DaprComponents_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DaprComponents"
+        ],
+        "summary": "Delete a Dapr Component.",
+        "description": "Delete a Dapr Component from a Managed Environment.",
+        "operationId": "DaprComponents_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "description": "Name of the Managed Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the Dapr Component.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Delete operation completed"
+          },
+          "204": {
+            "description": "Environment does not exist"
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete dapr component": {
+            "$ref": "./examples/DaprComponents_Delete.json"
+          }
+        },
+        "x-ms-long-running-operation": false
+      }
+    }
+  },
+  "definitions": {
+    "DaprComponentsCollection": {
+      "description": "Dapr Components ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DaprComponent"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DaprComponent": {
+      "description": "Dapr Component.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
```

Comment:
PR looks good to me I am signing off. Only thing I would suggest is to consider if this needs to be a tracked resource. The disadvantage of having proxy resources is that they do not play very well with azure RBAC and will only directly inherit thier permissions from the parent resource. If this works for your scenario then you can go ahead , but if you think you need to provide your customers with more fine grained RBAC on individual "daprcomponent" resource then this is something to consider.